src/signals.rs

```diff
@@ -9,6 +9,21 @@ pub(crate) fn ctrlc_protection(engine_state: &mut EngineState) {
     engine_state.set_signals(Signals::new(interrupt.clone()));
 
     let signal_handlers = Handlers::new();
+
+    // Register a handler to kill all background jobs on interrupt.
+    signal_handlers
+        .register_unguarded({
+            let jobs = engine_state.jobs.clone();
+            Box::new(move |action| {
+                if action == SignalAction::Interrupt {
+                    if let Ok(mut jobs) = jobs.lock() {
+                        jobs.kill_all().ok();
```

Comment:
also let's make this a bit more explicit
```suggestion
                        let _ = jobs.kill_all();
```